File: backend/app/schemas/analytics.py
from datetime import datetime

from pydantic import BaseModel


class AnalyticsSummary(BaseModel):
    row_count: int
    column_count: int


class ColumnAnalytics(BaseModel):
    name: str
    type: str
    null_count: int
    null_percent: float
    metrics: dict


class AnalyticsResponse(BaseModel):
    dataset_id: str
    computed_at: datetime
    summary: AnalyticsSummary
    columns: list[ColumnAnalytics]


class ApiResponse(BaseModel):
    success: bool = True
    data: dict | list | None = None
    error: dict | None = None

File: frontend/src/types/analytics.ts
export interface AnalyticsSummary {
  row_count: number;
  column_count: number;
}

export interface ColumnAnalytics {
  name: string;
  type: string;
  null_count: number;
  null_percent: number;
  metrics: Record<string, unknown>;
}

export interface AnalyticsProfile {
  dataset_id: string;
  computed_at: string;
  summary: AnalyticsSummary;
  columns: ColumnAnalytics[];
}

export interface NumericMetrics {
  sum?: number;
  avg?: number;
  min?: number;
  max?: number;
  count?: number;
  std_dev?: number;
}

export interface DateMetrics {
  min?: string;
  max?: string;
  count?: number;
}

export interface CategoricalMetrics {
  unique_count?: number;
  top_values?: { value: string; count: number }[];
}
